components/org.wso2.micro.integrator.initializer/src/main/java/org/wso2/micro/integrator/initializer/deployment/application/deployer/CappDeployer.java

```diff
@@ -220,39 +252,63 @@ private void deployCarbonApps(String artifactPath) throws CarbonException {
         }
     }
 
+    public void deployCarbonAppsDirectory() {
+
+        File cAppDirectory = new File(this.appsDir);
+        File[] cAppFiles = cAppDirectory.listFiles();
+        if (cAppFiles == null) {
```

Comment:
this is handled in for each right ?

Reply:
As per javadoc `The array will be empty if the directory is empty. Returns null if this abstract pathname does not denote a directory, or if an I/ O error occurs.`. 

This particular null check is to handle the null. Empty array will be handled with the below for-loop.